src/register/RegistrationPage.jsx

```diff
@@ -1,158 +1,131 @@
-  } = props;
+    thirdPartyAuthContext: {
+      autoSubmitRegForm,
+      errorMessage: thirdPartyAuthErrorMessage,
+      finishAuthUrl,
+      currentProvider,
+      providers,
+      secondaryProviders,
+      pipelineUserDetails,
+    },
+  } = useSelector(state => state.commonComponents);
 
-  const { formatMessage } = useIntl();
-  const countryList = useMemo(() => getCountryList(getLocale()), []);
+  const backendValidations = useMemo(
+    () => getBackendValidations(registrationError, validations), [registrationError, validations],
+  );
   const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
-  const registrationEmbedded = isHostAvailableInQueryParams();
-  const { cta, host } = queryParams;
   const tpaHint = useMemo(() => getTpaHint(), []);
-  const flags = {
-    showConfigurableEdxFields: getConfig().SHOW_CONFIGURABLE_EDX_FIELDS,
-    showConfigurableRegistrationFields: getConfig().ENABLE_DYNAMIC_REGISTRATION_FIELDS,
-    showMarketingEmailOptInCheckbox: getConfig().MARKETING_EMAILS_OPT_IN,
-  };
+  const { cta, host } = queryParams;
 
   const [formFields, setFormFields] = useState({ ...backedUpFormData.formFields });
   const [configurableFormFields, setConfigurableFormFields] = useState({ ...backedUpFormData.configurableFormFields });
   const [errors, setErrors] = useState({ ...backedUpFormData.errors });
-  const [emailSuggestion, setEmailSuggestion] = useState({ ...backedUpFormData.emailSuggestion });
-  const [autoSubmitRegisterForm, setAutoSubmitRegisterForm] = useState(false);
   const [errorCode, setErrorCode] = useState({ type: '', count: 0 });
   const [formStartTime, setFormStartTime] = useState(null);
-  const [focusedField, setFocusedField] = useState(null);
+  // temporary error state for embedded experience because we don't want to show errors on blur
+  const [temporaryErrors, setTemporaryErrors] = useState({ ...backedUpFormData.errors });
 
-  const {
-    providers, currentProvider, secondaryProviders, finishAuthUrl,
-  } = thirdPartyAuthContext;
-  const platformName = getConfig().SITE_NAME;
-  const buttonLabel = cta ? formatMessage(messages['create.account.cta.button'], { label: cta }) : formatMessage(messages['create.account.for.free.button']);
-
-  /**
-   * If auto submitting register form, we will check tos and honor code fields if they exist for feature parity.
-   */
-  const checkTOSandHonorCodeFields = () => {
-    if (Object.keys(fieldDescriptions).includes(FIELDS.HONOR_CODE)) {
-      setConfigurableFormFields(prevState => ({
-        ...prevState,
-        [FIELDS.HONOR_CODE]: true,
-      }));
-    }
-    if (Object.keys(fieldDescriptions).includes(FIELDS.TERMS_OF_SERVICE)) {
-      setConfigurableFormFields(prevState => ({
-        ...prevState,
-        [FIELDS.TERMS_OF_SERVICE]: true,
-      }));
-    }
-  };
+  const buttonLabel = cta
+    ? formatMessage(messages['create.account.cta.button'], { label: cta })
+    : formatMessage(messages['create.account.for.free.button']);
```

Comment:
Similarly for this, can this also be moved next to other constants?

Reply:
this is also dependent. Detailed answer above ^

Reply:
Maybe we can move cta and host const next to it.